src/python/pants/engine/process.py

```diff
@@ -313,7 +319,10 @@ class InteractiveRunner:
     _scheduler: "SchedulerSession"
 
     def run(self, request: InteractiveProcess) -> InteractiveProcessResult:
-        with ExceptionSink.ignoring_sigint():
+        if request.capture_signals:
+            with ExceptionSink.ignoring_sigint():
+                return self._scheduler.run_local_interactive_process(request)
+        else:
             return self._scheduler.run_local_interactive_process(request)
```

Comment:
You can remove this `else` block. The `if` block will early return. Might make this a little simpler to follow.